doc/reference/backends.rst

```diff
@@ -2,18 +2,32 @@
 Backends and Configs
 ********************
 
-.. note:: Both NetworkX backend and config systems are experimental. They
-   let you execute an alternative backend implementation instead of NetworkX's
-   pure Python dictionaries implementation. Things will almost certainly
-   change and break in future releases!
+.. note:: Both NetworkX backend and config systems are experimental.
 
+   Backends let you execute an alternative backend implementation instead of NetworkX's
+   pure Python dictionaries implementation. The config system provides library level
+   storage of configuration settings that otherwise might come from environment
+   variables.
+
+   Both systems are new and under active development. Hence, they will be regularly
+   updated with new features and improvements. At this stage, there are no deprecation
+   warnings for these systems. This may change as the features mature and become more
+   stable.
+
+   Users of these systems should expect minimal need for code updates to
+   maintain compatibility. However, staying informed about new features and changes
+   will be beneficial. Developers working on these systems should monitor changes
+   closely. Participating in weekly NX-dispatch meetings is an excellent way to stay
+   updated and contribute to discussions about ongoing developments.
```

Comment:
This still feels like a lot of cautionary statements that could generally apply to other parts of NetworkX itself, but because they're mentioned here it makes dispatching seem especially unstable IMO.  My preference would be to remove these two paragraphs.

Possibly a separate topic: why wouldn't we start adding deprecation warnings if we have to make user-facing interface changes to dispatching and/or config?

Reply:
If we didn't deprecate the name changes to e.g. the environmental variables, we probably should. We phased them out, but added code to allow them to work even though they aren't advertised in the docs anymore. At some point it would be good to remove that extra code. So a deprecation seems natural. 

Actually, I think we've maintained backward compatibility despite the many changes being made.

For this section, how about something like:

```
    Backends let you execute an alternative backend implementation instead of NetworkX's
    pure Python dictionaries implementation. Configs provides library level storage
    of configuration settings that otherwise might come from environment variables.

.. note:: Both NetworkX backend and config systems are new and under active
    development. Hence, they are regularly updated with new features and
    improvements. While the end user interface is unlikely to make breaking
    changes going forward, backend developers may see some need for code
    updates (with deprecation periods). Developers creating backends should
    monitor changes. Participating in weekly NX-dispatch meetings is an
    excellent way to contribute to discussions about ongoing developments.
```
(maybe with a link to the scientific-python calendars page?)